### ISSUE TITLE
3.0.3 broke my configuration

### ISSUE THREAD

**Issue by ivarconr** [log4js-node/log4js-node#768]
In version 3.0.2 this was legal configuration:

```js
const log4js = require('log4js');
const level = log4js.levels.ERROR;
log4js.configure({
        appenders: {
            console: { type: 'console' },
        },
        categories: {
            default: { appenders: ['console'], level },
        },
});
```

With log4js version 3.0.3 I get this error-message for the above config:
```
Error: Problem with log4js configuration: ({ appenders: { console: { type: 'console' } },
  categories: 
   { default: 
      { appenders: [ 'console' ],
        level: { level: 10000, levelStr: 'DEBUG', colour: 'cyan' } } } }) - category "default" is not valid (level "[object Object]" not recognised; valid levels are ALL, TRACE, DEBUG, INFO, WARN, ERROR, FATAL, MARK, OFF)
    at tests.forEach (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/configuration.js:25:13)
    at Array.forEach (<anonymous>)
    at Object.throwExceptionIf (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/configuration.js:23:9)
    at categoryNames.forEach (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/categories.js:53:19)
    at Array.forEach (<anonymous>)
    at configuration.addListener (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/categories.js:22:17)
    at listeners.forEach.listener (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/configuration.js:36:33)
    at Array.forEach (<anonymous>)
    at Object.configure (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/configuration.js:36:13)
    at Object.configure (/home/ivaosthu/code/unleash/unleash/node_modules/log4js/lib/log4js.js:65:17)
```

The workaround for now is to use `log4js.levels.ERROR.levelStr`, but it would be preferable to use the `level` object directly

**Comment by victor0801x** [comment]
not sure why you use level object directly ? it's not documented.

```js
// https://github.com/log4js-node/log4js-node/blob/master/lib/log4js.js#L64
const clonedConfigObject = CircularJSON.parse(CircularJSON.stringify(configObject));

// it seem `categorise.js` doesnot try to change categorise's config
// can fix by following
const { appenders, categories } = configObject;
const clonedAppenders = CircularJSON.parse(CircularJSON.stringify(appenders));
configuration.configure({ appenders: clonedAppenders, categories });
```

**Comment by ivarconr** [comment]
To let me not worry about typos and uppercase vs. lowercase.

**Comment by nomiddlename** [comment]
Fix published to npm in version 3.0.4
